### PR TITLE
remove params from subworkflows

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/prepare_intervals/main.nf [hunk 1]
```diff
@@ -26,29 +29,31 @@ workflow PREPARE_INTERVALS {
...
-        file("${params.outdir}/no_intervals.bed.gz").text     = "no_intervals\n"
-        file("${params.outdir}/no_intervals.bed.gz.tbi").text = "no_intervals\n"
-
-        intervals_bed        = Channel.fromPath(file("${params.outdir}/no_intervals.bed")).map{ it -> [ it, 0 ] }
-        intervals_bed_gz_tbi = Channel.fromPath(file("${params.outdir}/no_intervals.bed.{gz,gz.tbi}")).collect().map{ it -> [ it, 0 ] }
-        intervals_combined   = Channel.fromPath(file("${params.outdir}/no_intervals.bed")).map{ it -> [ [ id:it.simpleName ], it ] }
-    } else if (params.step != 'annotate' && params.step != 'controlfreec') {
+        file("${outdir}/no_intervals.bed").text        = "no_intervals\n"
+        file("${outdir}/no_intervals.bed.gz").text     = "no_intervals\n"
+        file("${outdir}/no_intervals.bed.gz.tbi").text = "no_intervals\n"
```
Comment (FriederikeHanssen) [inline subworkflows/local/prepare_intervals/main.nf:34]: I am wondering actually if we could dump this into the work dir in the long run.

File: workflows/sarek.nf [hunk 2]
```diff
@@ -83,14 +83,9 @@ for (param in checkPathParamList) if (param) file(param, checkIfExists: true)
 */
 
 // Initialize file channels based on params, defined in the params.genomes[params.genome] scope
-ascat_alleles           = params.ascat_alleles           ? Channel.fromPath(params.ascat_alleles).collect()           : Channel.empty()
-ascat_loci              = params.ascat_loci              ? Channel.fromPath(params.ascat_loci).collect()              : Channel.empty()
-ascat_loci_gc           = params.ascat_loci_gc           ? Channel.fromPath(params.ascat_loci_gc).collect()           : Channel.value([])
-ascat_loci_rt           = params.ascat_loci_rt           ? Channel.fromPath(params.ascat_loci_rt).collect()           : Channel.value([])
```
Comment (FriederikeHanssen) [inline workflows/sarek.nf:89]: I see what you are doing. I would love if we can find a solution in the future where the file channel building happens in one place and not scattered.

File: tests/test_intervals.yml [hunk 3]
```diff
@@ -138,8 +138,6 @@
     # binary changes md5sums on reruns
     - path: results/preprocessing/recalibrated/test/test.recal.cram.crai
     # binary changes md5sums on reruns
-    - path: results/reference/intervals
-      should_exist: false
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply tests/test_intervals.yml:142]: yeah something weird is happening there, and I went done a rabbit hole and gave up and I'll fix this in future PR

File: workflows/sarek.nf (hunk 2 above)
Comment (maxulysse) [reply workflows/sarek.nf:89]: so do I

**Review by FriederikeHanssen — APPROVED** [review]
Looks super good

**Comment by maxulysse** [comment]
> Looks super good

Thanks

**Pull request merged**